tests/phpunit/TestFramework/Coverage/JUnit/JUnitTestCaseSorterTest.php

```diff
@@ -96,4 +100,113 @@ public function test_it_returns_unique_and_sorted_by_time_test_cases(): void
         $this->assertCount(3, $uniqueSortedFileNames);
         $this->assertSame('/path/to/test-file-3', $uniqueSortedFileNames[0]);
     }
+
+    public function test_it_has_correct_constants_for_bucket_sort(): void
+    {
+        $this->assertLessThan(
+            // Quicksort's average O(n log n)
+            JUnitTestCaseSorter::USE_BUCKET_SORT_AFTER * log(JUnitTestCaseSorter::USE_BUCKET_SORT_AFTER),
+            // Bucket Sort's average O(n + k)
+            JUnitTestCaseSorter::USE_BUCKET_SORT_AFTER + JUnitTestCaseSorter::BUCKETS_COUNT
+        );
+    }
+
+    public function test_it_sorts_correctly(): void
+    {
+        $uniqueTestLocations = self::makeTestLocationsArray();
+
+        // Sanity check
+        $this->assertNotTrue(self::orderConstraintsValid($uniqueTestLocations));
+
+        $sortedTestLocations = iterator_to_array(JUnitTestCaseSorter::sort($uniqueTestLocations));
+        $this->assertTrue(self::orderConstraintsValid($sortedTestLocations), 'Bucket sort failed order check');
+
+        // Another sanity check
+        $sortedTestLocations = self::quicksort($uniqueTestLocations);
+        $this->assertTrue(self::orderConstraintsValid($uniqueTestLocations), 'Quicksort failed order check');
+    }
+
+    public function test_it_sorts_faster_than_quicksort(): void
```

Comment:
So it basically makes sure this is indeed a performance improvement?

Reply:
Yes, that's right.